src/Pos/Ssc/Class/Storage.hs

```diff
@@ -43,17 +42,21 @@ class Ssc ssc =>
       SscGStateClass ssc where
     -- | Load global state from DB by recreating it from recent blocks.
     sscLoadGlobalState
-        :: (HasContext LrcContext m, MonadDB m, WithLogger m)
+        :: (MonadDB m, WithLogger m)
         => m (SscGlobalState ssc)
+    -- | Dump global state to DB.
+    sscGlobalStateToBatch :: SscGlobalState ssc -> Tagged ssc [SomeBatchOp]
```

Comment:
`SscGlobalState` is a type family, and it's not necessarily injective. If two different instances define `SscGlobalState Foo = ()` and `SscGlobaState Bar = ()`, then if you pass `()` to `sscGlobalStateToBatch`, it won't know which `ssc` to use.